bundle/jsky.app.ot/src/main/java/jsky/app/ot/tpe/GemsGuideStarWorker.java

```diff
@@ -116,24 +120,35 @@ public void setInterrupted() {
     }
 
     /**
+     * AGS failed to run, so clear the AGS results.
+     */
+    public void clearResults() {
+        apply(tpe.getContext(), 0.0, GuideGroup.AutomaticInitial());
+    }
+
+    /**
      * Apply the results of the guide star search to the current observation,
      * setting the selected position angle and guide probe.
      */
     public void applyResults(GemsGuideStars gemsGuideStars) {
-        applyResults(tpe.getContext(), gemsGuideStars);
+        apply(tpe.getContext(), gemsGuideStars.pa().toDegrees(), GuideGroup.createActive(gemsGuideStars.guideGroup().getAll()));
     }
 
-    public static void applyResults(final TpeContext ctx, final GemsGuideStars gemsGuideStars) {
-        SPInstObsComp inst = ctx.instrument().orNull();
+    private static void apply(final TpeContext ctx, final double paInDegrees, final GuideGroup autoGroup) {
+        final SPInstObsComp inst = ctx.instrument().orNull();
         if (inst != null) {
-            inst.setPosAngleDegrees(gemsGuideStars.pa().toDegrees());
+            inst.setPosAngleDegrees(paInDegrees);
             ctx.instrument().commit();
         }
 
-        TargetObsComp targetObsComp = ctx.targets().orNull();
+        final TargetObsComp targetObsComp = ctx.targets().orNull();
         if (targetObsComp != null) {
-            TargetEnvironment env = targetObsComp.getTargetEnvironment();
-            targetObsComp.setTargetEnvironment(env.setPrimaryGuideGroup(gemsGuideStars.guideGroup()));
+            final TargetEnvironment envOld = targetObsComp.getTargetEnvironment();
+            final GuideEnvironment gEnvOld = envOld.getGuideEnvironment();
+
+            final GuideEnvironment gEnvNew = gEnvOld.setGroup(0, autoGroup).setPrimaryIndex(0);
```

Comment:
Note: All of this is working now. However, if you have a manual group for GeMS and run manual search, by the way manual search works, it will overwrite your manual group. (Auto group will always be maintained, of course.)